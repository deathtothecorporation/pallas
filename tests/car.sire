;; TODO Need to fix this since pin shape has changed.

= (car x) | 1 _&4 (n a _ & 0 n a) (h _ & h) 0 x
= (cdr x) | 1 x&x (_ _ b & b)     (_ t & t) 0 x

= $4 (car car)

= tags ,, cdr | car | car | cdr car
       ,, cdr | car | car | cdr cdr

k4=(| cdr | car | car | car | cdr | car | cdr | cdr car)

= car (car-car ((car cdr-car) (cdr cdr-car)))

%[=car =cdr =4 =tags =k4]
