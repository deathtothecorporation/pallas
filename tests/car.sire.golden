(car a)=(1:(b&4):(b c d & 0-b-c):(b c & b) 0 a)
(cdr a)=(1:(b&b):(b c d & d):(b c & c) 0 a)
_=(%[4=0 k4=0 car=(0 0-car) =cdr tags=[0 0]])

_hash=(x#4056a18a552b3f589ff2067050a3d3f812909659073bcf1bce960866e5fc11cb)

